core-api/src/test/java/com/optimizely/ab/odp/ODPEventManagerTest.java

```diff
@@ -255,6 +255,157 @@ public void validateEventData() {
         assertFalse(event.isDataValid());
     }
 
+    @Test
+    public void validateAugmentCommonData() {
+        Map<String, Object> sourceData = new HashMap<>();
+        sourceData.put("k1", "source-1");
+        sourceData.put("k2", "source-2");
+        Map<String, Object> userCommonData = new HashMap<>();
+        userCommonData.put("k3", "common-1");
+        userCommonData.put("k4", "common-2");
+
+        Mockito.reset(mockApiManager);
+        ODPEventManager eventManager = new ODPEventManager(mockApiManager);
+        eventManager.setUserCommonData(userCommonData);
+
+        Map<String, Object> merged = eventManager.augmentCommonData(sourceData);
+
+        // event-sourceData
+        assertEquals(merged.get("k1"), "source-1");
+        assertEquals(merged.get("k2"), "source-2");
+        // userCommonData
+        assertEquals(merged.get("k3"), "common-1");
+        assertEquals(merged.get("k4"), "common-2");
+        // sdk-generated common data
+        assertNotNull(merged.get("idempotence_id"));
+        assertEquals(merged.get("data_source_type"), "sdk");
+        assertNotNull(merged.get("data_source"));
```

Comment:
can we verify here that the datasource is java sdk and in android we can add a unit test that will verify that it is android sdk?